src/net_processing.cpp

```diff
@@ -1605,7 +1623,15 @@ void static ProcessGetData(CNode* pfrom, const CChainParams& chainparams, CConnm
                     push = true;
                 }
             }
-            if (!push) {
+
+            if (push) {
```

Comment:
I don't understand why this condition matters.
I can think about 2 attack vectors:
1. Traditional privacy problem, where someone GETDATAs to probe if you have a transaction. If we remove the condition in this case, no extra privacy leak happens.
2. Where someone preventing you from rebroadcast by sending this unsolicited GETDATA. I'm not sure this is a valid concern and a practical scenario. Were you thinking about this?

Also, regrading where push is coming from: afaik mapRelay is not reliable (would it even work in rebroadcast case?), and the second condition is... complicated. 

Anyway, can we at least document why this flag is checked?